filament/src/ColorGrading.cpp

```diff
@@ -37,7 +41,44 @@ using namespace backend;
 
 static constexpr size_t LUT_DIMENSION = 32u;
 
-Tonemapper::Tonemapper(FEngine& engine) {
+struct ColorGrading::BuilderDetails {
+    ToneMapping mToneMapping = ToneMapping::ACES;
+};
+
+using BuilderType = ColorGrading;
+BuilderType::Builder::Builder() noexcept = default;
+BuilderType::Builder::~Builder() noexcept = default;
+BuilderType::Builder::Builder(BuilderType::Builder const& rhs) noexcept = default;
+BuilderType::Builder::Builder(BuilderType::Builder&& rhs) noexcept = default;
+BuilderType::Builder& BuilderType::Builder::operator=(BuilderType::Builder const& rhs) noexcept = default;
+BuilderType::Builder& BuilderType::Builder::operator=(BuilderType::Builder&& rhs) noexcept = default;
+
+ColorGrading::Builder& ColorGrading::Builder::toneMapping(ToneMapping toneMapping) noexcept {
+    mImpl->mToneMapping = toneMapping;
+    return *this;
+}
+
+ColorGrading* ColorGrading::Builder::build(Engine& engine) {
+    return upcast(engine).createColorGrading(*this);
+}
+
+std::function<float3(float3)> selectToneMapping(ColorGrading::ToneMapping toneMapping) {
```

Comment:
It could be slightly more efficient if we just used a pointer-to-member, I think std::function<> is pretty big and has to do some type erasing, which  in this case is not needed, since we know the type.
You could just have a `using TonemapperFunc = float3(ColorGRading::*)(float3);`

Reply:
Yeah I was hoping for some kind of small std::function optimization. But I can use a raw function pointer.